src/xenia/kernel/xnotifylistener.h

```diff
@@ -22,13 +22,13 @@ namespace xe {
 namespace kernel {
 
 union XNotificationKey {
+  XNotificationID id;
```

Comment:
This is not something obvious to rely on (and also you may want to initialize bit fields specifically instead), it's better to specify the intention to set the `id` clearly in the problematic place.

Reply:
Indeed. I also don't like the fact that it is opaque and required knowledge of this fact, but in terms of overall code structure it seems to be a better way to handle this.

Surely if this is such a problematic case then I can initialize it normally or even better create a constructor for it?

Reply:
Yes, a constructor (maybe not even `explicit` so XNotificationID can be passed to functions expecting XNotificationKey directly, but not sure) for creation from a complete notification ID would be nice, as this seems to be an important special case.